### PR TITLE
use puppetlabs-hocon for puppetserver.conf

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: manifests/server/puppetserver.pp [hunk 1]
```diff
@@ -245,9 +245,88 @@
...
+
+  hocon_setting { 'jruby-puppet.environment-class-cache-enabled':
+    ensure  => $envionment_class_cache_ensure,
+    path    => $puppetserver_conf,
+    setting => 'jruby-puppet.environment-class-cache-enabled',
+    value   => $server_environment_class_cache_enabled,
+    require => File[$puppetserver_conf],
+  }
+
+  # TODO: See https://tickets.puppetlabs.com/browse/HC-100
```
Comment (mmoll) [inline manifests/server/puppetserver.pp]: this is ugly, but I don't see that value change in reality, anyway

File: manifests/server/puppetserver.pp [hunk 2]
```diff
@@ -245,9 +245,88 @@
...
+      path    => $puppetserver_conf,
+      setting => $setting,
+      value   => $value,
+      type    => $type,
+      require => File[$puppetserver_conf],
+    }
+  }
+
+  if versioncmp($server_puppetserver_version, '2.3') >= 0 {
+    $envionment_class_cache_ensure = present
```
Comment (ekohl) [inline manifests/server/puppetserver.pp]: ```s/envionment/environment/``` (and below as well)

**Comment by ekohl** [comment]
What should we do with this? The issue was accepted by PL but ofc that's no ETA on a fix ...

**Comment by mmoll** [comment]
unsure, but I think there's no need to rush this in...

**Comment by ekohl** [comment]
In that case let's wait a bit.

**Pull request closed without merging**